.ci/install_kata_image.sh

```diff
@@ -19,151 +28,163 @@ ARCH="$(${cidir}/kata-arch.sh -d)"
 AGENT_INIT=${AGENT_INIT:-no}
 TEST_INITRD=${TEST_INITRD:-no}
 
-TMP_DIR=
-ROOTFS_DIR=
-
-PACKAGED_IMAGE="kata-containers-image"
-IMG_PATH="/usr/share/kata-containers"
-IMG_NAME="kata-containers.img"
-IMAGE_TYPE="assets.image.meta.image-type"
+IMAGE_DIR="/usr/share/kata-containers"
+IMG_LINK_NAME="kata-containers.img"
+INITRD_LINK_NAME="kata-containers-initrd.img"
+IMAGE_TYPE_KEY="assets.image.meta.image-type"
+IMAGE_OS_VERSION_KEY="assets.image.architecture.$(uname -m).version"
+
+if [ "${TEST_INITRD}" == "no" ]; then
+	OSBUILDER_YAML_INSTALL_NAME="osbuilder-image.yaml"
+	LINK_PATH="${IMAGE_DIR}/${IMG_LINK_NAME}"
+else
+	OSBUILDER_YAML_INSTALL_NAME="osbuilder-initrd.yaml"
+	LINK_PATH="${IMAGE_DIR}/${INITRD_LINK_NAME}"
+fi
 
 agent_path="${GOPATH}/src/github.com/kata-containers/agent"
-
-IMG_MOUNT_DIR=
-LOOP_DEVICE=
-
-# Build Kata agent
-bash -f "${cidir}/install_agent.sh"
-agent_commit=$(git --work-tree="${agent_path}" --git-dir="${agent_path}/.git" log --format=%h -1 HEAD)
-
-cleanup() {
-	[ -d "${ROOTFS_DIR}" ] && [[ "${ROOTFS_DIR}" = *"rootfs"* ]] && sudo rm -rf "${ROOTFS_DIR}"
-	[ -d "${TMP_DIR}" ] && rm -rf "${TMP_DIR}"
-	if [ -n "${IMG_MOUNT_DIR}" ] && mount | grep -q "${IMG_MOUNT_DIR}"; then
-		sudo umount "${IMG_MOUNT_DIR}"
-	fi
-	if [ -d "${IMG_MOUNT_DIR}" ]; then
-		rm -rf "${IMG_MOUNT_DIR}"
-	fi
-	if [ -n "${LOOP_DEVICE}" ]; then
-		sudo losetup -d "${LOOP_DEVICE}"
+osbuilder_repo="github.com/kata-containers/osbuilder"
+osbuilder_path="${GOPATH}/src/${osbuilder_repo}"
+latest_build_url="http://jenkins.katacontainers.io/job/image-nightly-$(uname -m)/lastSuccessfulBuild/artifact/artifacts"
+
+install_ci_cache_image() {
+	type=${1}
+	check_not_empty "$type" "image type not provided"
+	info "Install pre-built ${type}"
+	local image_name=$(curl -fsL "${latest_build_url}/latest-${type}")
+	sudo mkdir -p "${IMAGE_DIR}"
+	pushd "${IMAGE_DIR}" > /dev/null
+	local image_path=$(readlink -f "${IMAGE_DIR}/${image_name}")
+
+	sudo curl -fsOL "${latest_build_url}/shasum256-${type}-tarball"
```

Comment:
Do you really need to use `sudo` with a network command like this (same comment for instances below)?